File: fieldmate/src/main/java/com/nguyencong/fieldmate/service/SportTypeService.java
package com.nguyencong.fieldmate.service;

import java.util.List;

import com.nguyencong.fieldmate.dto.request.SportTypeRequest;
import com.nguyencong.fieldmate.dto.response.SportTypeResponse;

public interface SportTypeService {
    List<SportTypeResponse> getAllSportTypes();
    SportTypeResponse createSportType(SportTypeRequest request);
    SportTypeResponse updateSportType(Long id, SportTypeRequest request);
    void deleteSportType(Long id);
}

File: front-end/fieldmate-web/src/services/sport-type.service.ts
import {
  fieldmateClient,
  fieldmateEndpoints,
} from "@/configs/fieldmate-client";
import type {
  SportTypeRequest,
  SportTypeResponse,
} from "@/types/sport-type";

export const sportTypeService = {
  async getAll() {
    const response =
      await fieldmateClient.get<SportTypeResponse[]>(
        fieldmateEndpoints.sportTypes,
      );

    return response.data;
  },

  async create(request: SportTypeRequest) {
    const response = await fieldmateClient.post<SportTypeResponse>(
      fieldmateEndpoints.secureSportTypes,
      request,
    );

    return response.data;
  },

  async update(sportTypeId: number, request: SportTypeRequest) {
    const response = await fieldmateClient.put<SportTypeResponse>(
      fieldmateEndpoints.secureSportType(sportTypeId),
      request,
    );

    return response.data;
  },

  async remove(sportTypeId: number) {
    await fieldmateClient.delete(
      fieldmateEndpoints.secureSportType(sportTypeId),
    );
  },
};
